File: PseudoCoupHQ/Research/oracle/riscv/softfloat_slices/arch_units/go/au_212_c_alignof_gnu_f32.go
// arch-unit 212  --  c  `__alignof a`  lhs=float rhs=None
// symbol op_63   outcome LIFTED   2 arch-opcodes
//
// the arch-unit, arch-opcode by arch-opcode:
//   c.li a0, 0x4                         integer    constant
//   c.jr ra                              integer    return
//
// answer: uint64_t, 64 bits.  parameters are operand bit patterns.
package archunits

func Au_212_c_alignof_gnu_f32(p0 uint64) uint64 {
	// fa0: operand `a` (float) arrives in fa0 as a bit pattern
	var v1 uint64 = ((p0) & uint64(0xffffffff)); _ = v1
	var v2 uint64 = uint64(0x4); _ = v2
	// the answer is uint64_t, 64 bits
	return v2
}

File: PseudoCoupHQ/Research/oracle/riscv/softfloat_slices/arch_units/c/au_212_c_alignof_gnu_f32.c
/*
 * arch-unit 212  --  c  `__alignof a`  lhs=float rhs=None
 * symbol op_63   outcome LIFTED   2 arch-opcodes
 *
 * the arch-unit, arch-opcode by arch-opcode:
 *   c.li a0, 0x4                         integer    constant
 *   c.jr ra                              integer    return
 *
 * answer: uint64_t, 64 bits.  parameters are operand bit patterns.
 */
#include "arch_units_int.h"

uint64_t au_212_c_alignof_gnu_f32(uint64_t p0)
{
    /* fa0: operand `a` (float) arrives in fa0 as a bit pattern */
    const uint64_t v1 = ((p0) & UINT64_C(0xffffffff));
    const uint64_t v2 = UINT64_C(0x4);
    /* the answer is uint64_t, 64 bits */
    return v2;
}
